app/uk/gov/hmrc/individualincomedesstub/util/EmployerReferenceStringBinder.scala

```diff
@@ -16,18 +16,17 @@
 
 package uk.gov.hmrc.individualincomedesstub.util
 
-import java.net.URLDecoder
+import uk.gov.hmrc.domain.EmpRef
 
-import uk.gov.hmrc.individualincomedesstub.domain.EmployerReference
+class EmployerReferenceStringBinder extends AbstractPathStringBindable[EmpRef] {
 
-class EmployerReferenceStringBinder extends AbstractPathStringBindable[EmployerReference] {
+  def bind(key: String, value: String) =
```

Comment:
Now that there's a second implementation of `AbstractPathStringBindable[T]`, then would go a step further here and abstract the above `Try(...) getOrElse (...)` into `AbstractPathStringBindable`:
```
  def bind[T](key: String, value: String, message: String, function: => T): Either[String, T] =
    Try(Right(function)) getOrElse Left(errorResponse(message))
```
The end result is a less verbose `bind(String, String)` implementation here in `EmployerReferenceStringBinder`:
```
  override def bind(key: String, value: String): Either[String, EmpRef] =
    bind(key, value, "Invalid employer reference submitted", EmpRef.fromIdentifiers(value))
```
Similarly in `NinoPathStringBinder` it would then end up with a one-liner:
```
  override def bind(key: String, value: String): Either[String, Nino] =
    bind(key, value, "Malformed nino submitted", Nino(value))
```